src/main/java/org/terasology/gf/tree/lsystem/AdvancedLSystemTreeDefinition.java

```diff
@@ -345,16 +330,16 @@ private TreeStructure generateTreeFromAxion(Vector3i location, String currentAxi
         Deque<BranchLocation> stackBranch = Queues.newArrayDeque();
 
         BranchLocation branchLocation = treeStructure.getRootBranch();
-        Vector3f position = location.toVector3f();
-        Matrix4f rotation = new Matrix4f(new Quat4f(new Vector3f(0, 1, 0), treeRotation), BaseVector3f.ZERO, 1.0f);
+        Vector3f position = new Vector3f(location);
+        Matrix4f rotation = new Matrix4f().translationRotateScale(new Vector3f(),new Quaternionf().setAngleAxis(treeRotation,0,1,0), 1.0f);
```

Comment:
I think @pollend told me to just use a quaternion for bascially the same code in `RecursiveTreeGeneratorLSystem`... see https://github.com/Terasology/CoreWorlds/commit/6484ebc671bf27511a7eebceeb67e19968d7f33c

Reply:
However, it's not as straight forward, and this code isn't working properly anyways right now. Will add a TODO comment to pick this up later.